openhands/llm/llm.py

```diff
@@ -216,6 +217,7 @@ def wrapper(*args, **kwargs):
             temperature=self.config.temperature,
             top_p=self.config.top_p,
             drop_params=True,
+            caching_prompt=self.config.caching_prompt,
```

Comment:
This should **not** use the `config` value, but then also be changed to `self.is_caching_active()` (per my other comment)

Reply:
thanks for your suggestion, using `self.is_caching_active()` is indeed more intuitive.